File: php/difference-of-squares/DifferenceOfSquares.php
<?php declare(strict_types=1);

function squareOfSum(int $max): int
{
    $r = 0;
    for ($i = 0; $i <= $max; $i++) {
        $r += $i;
    }
    return $r ** 2;
}

function sumOfSquares(int $max): int
{
    $r = 0;
    for ($i = 0; $i <= $max; $i++) {
        $r += $i**2;
    }
    return $r;
}

function difference(int $max): int
{
    return squareOfSum($max) - sumOfSquares($max);
}

File: swift/difference-of-squares/Sources/DifferenceOfSquares/DifferenceOfSquares.swift
class Squares {
  var squareOfSum: Int
  var sumOfSquares: Int
  var differenceOfSquares: Int
  
  // Write your code for the 'Difference Of Squares' exercise here.
  init(_ max: Int) {
    squareOfSum = 0
    for i in 1...max {
      squareOfSum += i
    }
    squareOfSum *= squareOfSum
    
    sumOfSquares = 0
    for i in 1...max {
      sumOfSquares += i * i
    }

    differenceOfSquares = squareOfSum - sumOfSquares
  }
}
